crates/tests/sawtooth_publisher/src/main.rs

```diff
@@ -63,31 +59,45 @@ pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
         height * iterations
     );
     let c8y_msg = Topic::new("tedge/measurements")?;
-    let c8y_cmd = Topic::new("c8y/s/ds")?;
-    let c8y_err = Topic::new("c8y/s/e")?;
+    let c8y_err = TopicFilter::new("c8y/s/e")?;
 
     init_logger();
 
     let name = "sawtooth_".to_string() + &process::id().to_string();
-    let mqtt = Client::connect(&name, &Config::default()).await?;
+    let config = Config::default()
+        .with_clean_session(true)
+        .with_session_name(name)
+        .with_subscriptions(c8y_err);
+    let mqtt = Connection::new(&config).await?;
 
-    let commands = mqtt.subscribe(c8y_cmd.filter()).await?;
-    let c8y_errors = mqtt.subscribe(c8y_err.filter()).await?;
-    let errors = mqtt.subscribe_errors();
+    let c8y_messages = mqtt.published;
+    let c8y_errors = mqtt.received;
+    let errors = mqtt.errors;
 
     let start = Instant::now();
 
     if template == "flux" {
-        tokio::spawn(publish_topic(mqtt, c8y_msg, wait, height, iterations));
+        tokio::spawn(publish_topic(
+            c8y_messages,
+            c8y_msg,
```

Comment:
The c8y_messages and c8y_msg are a bit confusing. Why do we need both?

Reply:
Agree. The latter is the topic, the former the stream of messages.

I will come with better names.